### ISSUE TITLE
A slightly different approach for plot.sfnetwork

### ISSUE THREAD

**Issue by agila5** [luukvdmeer/sfnetworks#226]
**Is your feature request related to a problem? Please describe.**
I think that the current implementation of `plot.sfnetwork` is slightly inefficient when the input `x` contains spatially explicit edges (or `draw_lines = TRUE`). The problem is that, in those cases, `dots$x` mixes `POINT` and `LINESTRING` objects, which implies that `do.call(plot, dots)` actually calls `sf:::plot.sfc_GEOMETRY`. 

`sf:::plot.sfc_GEOMETRY` is less efficient than just calling `sf:::plot.sfc_POINT` and `sf:::plot.sfc_LINESTRING` separately since `sf:::plot.sfc_GEOMETRY` [plots one node at a time](https://github.com/r-spatial/sf/blob/92438cd8d989bba772cb3a6604f007f54dc837ce/R/plot.R#L514-L516) wheteas `sf:::plot.sfc_POINT` [plots all points together](https://github.com/r-spatial/sf/blob/92438cd8d989bba772cb3a6604f007f54dc837ce/R/plot.R#L316-L320). 

**Describe the solution you'd like**
Two separate calls to `do.call` to plot one set of features at a time. If you agree, I will prepare a PR. 

**Describe alternatives you've considered**
We can manually extract and plot nodes/lines, but that's quite boring.

**Comment by loreabad6** [comment]
I totally agree, the plotting method is quite slow. Thank you @agila5 for preparing a PR.
